functions/main.py

```
@@ -401,38 +352,48 @@ def _extract_text_from_file(file: FileStorage) -> str:
     if req.method != 'POST':
         return https_fn.Response("Method not allowed", status=405)
-    
+
+    # Verify Firebase Auth token
+    auth_header = req.headers.get('Authorization')
+    if not auth_header or not auth_header.startswith('Bearer '):
+        return https_fn.Response("Unauthorized", status=401)
+
+    id_token = auth_header.split('Bearer ')[1]
+    try:
+        decoded_token = auth.verify_id_token(id_token)
```

Ruff (0.12.2)

Local variable `decoded_token` is assigned to but never used

Remove assignment to unused variable `decoded_token`

(F841)